### ISSUE TITLE
When changing a song's artist, writing to id3 tags fails.

### ISSUE THREAD

**Issue by wagnered** [ampache/ampache#2460]
Didn't get a chance to look closer: I started looking into a possible race condition when changing artists.

song.class) -> Writing id3 metadata to file /home/music/Emerson, Lake & Palmer [Discography]/1994 - In The Hot Seat/11-Pictures At An Exhibition.mp3 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> Failed to write tags! ID3v2: Skipping "FILE" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "BITRATE" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "RATE" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MODE" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MIME" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "ALBUMARTIST" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MB_TRACKID" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MB_ALBUMID" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MB_ARTISTID" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "MB_ALBUMARTISTID" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "TRACK" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "REPLAYGAIN_TRACK_GAIN" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "REPLAYGAIN_TRACK_PEAK" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "REPLAYGAIN_ALBUM_GAIN" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> ID3v2: Skipping "REPLAYGAIN_ALBUM_PEAK" because cannot match it to a known ID3v2 frame type 
2020-07-21T09:39:16-07:00 [admin] (vainfo.class) -> WriteID3v2() failed with message(s):<pre><ul><li>Only one COMM tag allowed with the same Language + Description (eng + )</li><li>Frame &quot;COMM&quot; is NOT allowed</li><li>$this-&gt;GenerateID3v2FrameData() failed for &quot;COMM&quot;</li><li>$this-&gt;GenerateID3v2Tag() failed</li></ul></pre> 
2020-07-21T09:39:17-07:00 [admin] (vainfo.class) -> Cleaning id3v1 
2020-07-21T09:39:17-07:00 [admin] (vainfo.class) -> Cleaning id3v2

**Comment by lachlan-00** [comment]
good to know, i'll need to enable this too

**Comment by lachlan-00** [comment]
I think I'm getting something similar adding tags/genre

**Comment by wagnered** [comment]
Some more.
ID3v2 TXXX data is not properly structured 

Skipping "RECORDING_TIME" because cannot match it to a known ID3v2 frame type.

Also, possibly related to the album I just added: 

`
2020-07-22T21:49:57-07:00 [admin] (browse.class) -> Show objects called for type {song} 
2020-07-22T21:49:58-07:00 [ampache] (art.class) -> Error: No mime type found. 
`

It's not a fatal error and don't know what the lack of a mime type will do or not do so I didn't make a new bug issue.

**Comment by lachlan-00** [comment]
it looks to me like that's what happens with a blank/nonimage for a new album?
 
![image](https://user-images.githubusercontent.com/1305249/88256239-16881d00-ccfe-11ea-9e1d-33a68a11ab57.png)

**Comment by lachlan-00** [comment]
i put some debug on the end of that to print out the mime that was inserted. it's doing a regex with whatever is being put in to get a mime_type

![image](https://user-images.githubusercontent.com/1305249/88256374-8696a300-ccfe-11ea-9655-94d37aab8c7a.png)

**Comment by wagnered** [comment]
What else have you done?🙂

The reason I ask:. I tried stopping the catalog add process by stopping the lighttpd web browser.  During the time that it is stopped, the log shows no more activity.  But when I start the web browser, the log reports continued catalog processing.  Actually it reports more art gathering.  How can it continue after the browser/PHP has been stopped?

The reason I tried to stopped it was because the browser response was very slow but I might have ajax disabled.

Anyway, some of my mp3s have the TXXX tag so now writing ID3v2.3 includes those along with one album cover and one artist image.

**Comment by lachlan-00** [comment]
stebe had an issue probably a year ago where the UI processes were restarting continually but i think that was ajax related

**Comment by lachlan-00** [comment]
i just tested it and it restarted again too, why that's happened i got no idea, i restarted fpm and it restarted the add while tailing the log

Logging out stopped the process restarting though.

**Comment by wagnered** [comment]
That's good to know.  I'm using lighttpd which loads mod_fastcgi.  I think that fastcgi and fpm run their processes independently so killing the server doesn't do anything and  I'm guessing that closing the session causes the browser to signal fastcgi/fpm to stop.

**Comment by lachlan-00** [comment]
i have switched to fpm with apache for a long time that would make sense, i wonder if we need to restart the processes for catalog updates at all?

**Comment by wagnered** [comment]
I'm not sure what situation **might** require restarting a catalog update, other than logging out.  That's an interesting question.  Under normal circumstances PHP time-out is disabled.  So unless the process runs out of memory  while processing there should be no problem.  Although I seem to remember that when I had over 30000-40000 files or maybe more, I had to restart it to complete the catalog. Maybe it had something to with the duration of the script.

**Comment by lachlan-00** [comment]
that's true it should obviously catch up after adding 20k files much faster so should be able to complete with a restart in that example. makes sense to let people do it that way.

i remember back when stebe had the problem it was due to a bug in the process causing it to restart like crazy.

**Comment by stebe** [comment]
I still have the restarting problem. Still can't pinpoint why. This has been happening before and after I switched to fpm. Logging out is how I get around this.

**Comment by wagnered** [comment]
According to PHP.net [fastcgi-finish-request](https://www.php.net/manual/en/function.fastcgi-finish-request.php), the only fpm user function, may be a way to stop the process.  It should be used with the caveat that session_write_close should be issued just before `fastcgi-finish-request` and using it excessively can use up all the fpm threads.
